desc/objectives/_coils.py

```diff
@@ -1267,6 +1283,197 @@ def compute(self, field_params, constants=None):
         return f
 
 
+class QuadraticFluxMinimizingSurface(_Objective):
+    """Target B*n = 0 on a surface.
+
+    Used to find a quadratic-flux-minimizing (QFM) surface, so a
+    `FourierRZToroidalSurface` should be passed to the objective.
+    Should always be used along with a ``ToroidalFlux`` objective to
+    ensure that the resulting QFM surface has the desired amount of
+    flux enclosed and avoid trivial solutions.
+
+    Parameters
+    ----------
+    surface :  FourierRZToroidalSurface
+        QFM surface upon which the normal field error will be minimized.
```

Comment:
This is a dummy question but I just ask it to learn these kind of stuff. So, `QuadraticFlux` objective takes an equilibrium and computes the `B dot n` error on only the LCFS (then this keeps the equilibrium fixed and changes the coils to minimize the error). This new one though, takes a surface and minimizes the error on that. This could be either LCFS or not. Am I correct?

The part I am confused though is if the surface is also kept fixed or not? Docs of `ToroidalFlux` says that surface may vary.

Reply:
And is this anyway related to free boundary objective?

Reply:
Ah yes I need to change ToroidalFlux to allow the thing to vary...

QFM the surface can vary.

It is sort of related, in that for vacuum free bdry techincally if you match the toroidal flux and the B dot n error is zero, that surface is the LCFS surface for a free bdry equilibrium (assuming the true field inside has nested surfaces).

Some people use QFM surfaces instead of doing full free bdry when they do single stage optimization for this reason, I believe, it is cheaper than doing actual free bdry